nodes/core/io/10-mqtt.js

```diff
@@ -86,6 +86,9 @@ module.exports = function(RED) {
 
         // Create the URL to pass in to the MQTT.js library
         if (this.brokerurl === "") {
+            if (this.broker.indexOf("//") > -1) {
+                this.brokerurl = this.broker;
+            } else {
```

Comment:
This effectively means the Port setting is going to be ignored... but it is a required field so the user has to provide something. That will be confusing.